src/modules/AutoinstFunctions.rb

```diff
@@ -81,7 +81,8 @@ def check_second_stage_environment
     # 2) impllicitly according to software selection
     # 3) if not set explicitly and just one product is available on media - use it
     #
-    # @return [Y2Packager::Product] a base product or nil
+    # @return [Y2Packager::Product|Y2Packager::ProductLocation] a base product or nil
```

Comment:
I would separate both sentences: "... a base product or nil. The returned class depends..."